dali/python/nvidia/dali/pipeline.py

```diff
@@ -1484,6 +1500,20 @@ def _generate_build_args(self):
                 for (name, dev), dtype, ndim in zip(self._names_and_devices, dtypes, ndims)]
 
 
+def _shutdown_pipelines():
+    for weak in list(Pipeline._pipes):
+        p = weak()
+        if p is None:
+            Pipeline._pipes.discard(weak)
+            continue
+        p._shutdown()
```

Comment:
Will that work as well?
```suggestion
        if p is None:
            p._shutdown()
```